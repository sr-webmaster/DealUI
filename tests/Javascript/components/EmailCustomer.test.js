import { mount } from '@vue/test-utils'
import Component from '~/components/EmailCustomer.vue'

describe('Email Customer', () => {
  it('snapshot default', () => {
    const user = {
      id: 999,
      name: 'User',
      email: 'info@mail.com',
      phone: '766-888-9999'
    }
    const templates = [
      'Template One',
      'Template Two'
    ]
    const deal = {
      id: 1,
      name: 'Your Best Chance'
    }

    const wrapper = mount(Component, {
      propsData: {
        dialog: false,
        user: user,
        templates: templates,
        deal: deal
      }

    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.text()).toContain(user.email)
    expect(wrapper.props().user).toBe(user)
    expect(wrapper.props().deal).toBe(deal)
    expect(wrapper.props().templates).toBe(templates)
    wrapper.findAll('button').trigger('click')
    expect(wrapper.emitted('close')).toHaveLength(2)
    expect(wrapper.emitted('save')).toHaveLength(1)
  })
})
