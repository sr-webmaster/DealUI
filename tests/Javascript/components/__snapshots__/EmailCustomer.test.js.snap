// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Customer snapshot default 1`] = `
<div
  class="container bgwhite"
>
  <div
    class="layout row wrap"
  >
    <div
      class="flex d-flex"
    >
      <div>
        <div
          class="layout row justify-space-between"
        >
          <h2>
            Email Customer
          </h2>
           
          <button
            class="mr-0 mt-0 small-btn v-btn v-btn--icon theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              <i
                aria-hidden="true"
                class="v-icon grey--text fas fa-times theme--light"
              />
            </div>
          </button>
        </div>
         
        <div
          class="layout row"
        >
          <div
            class="flex d-flex"
          >
            <div
              class="v-card v-card--flat v-sheet v-sheet--tile theme--light"
            >
              <div
                class="v-card__title"
              >
                <div>
                  <div
                    class="subheading"
                  >
                    
                    Customer
                  
                  </div>
                   
                  <span
                    class="body-2"
                  >
                    User
                  </span>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="flex d-flex"
          >
            <div
              class="v-card v-card--flat v-sheet v-sheet--tile theme--light"
            >
              <div
                class="v-card__title"
              >
                <div>
                  <div
                    class="subheading"
                  >
                    
                    Customer ID
                  
                  </div>
                   
                  <span
                    class="body-2"
                  >
                    999
                  </span>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="flex d-flex"
          >
            <div
              class="v-card v-card--flat v-sheet v-sheet--tile theme--light"
            >
              <div
                class="v-card__title"
              >
                <div>
                  <div
                    class="subheading"
                  >
                    
                    Email Address
                  
                  </div>
                   
                  <span
                    class="body-2"
                  >
                    info@mail.com
                  </span>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="flex d-flex"
          >
            <div
              class="v-card v-card--flat v-sheet v-sheet--tile theme--light"
            >
              <div
                class="v-card__title"
              >
                <div>
                  <div
                    class="subheading"
                  >
                    
                    Phone
                  
                  </div>
                   
                  <span
                    class="body-2"
                  >
                    766-888-9999
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="flex d-flex justify-end xs12"
    >
      <div>
        <form
          class="v-form"
          novalidate="novalidate"
        >
          <div
            class="v-input v-text-field v-text-field--enclosed v-text-field--outline v-input--is-label-active v-input--is-dirty v-input--is-disabled theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active v-label--is-disabled theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Opportunity
                  </label>
                  <input
                    aria-label="Opportunity"
                    disabled="disabled"
                    required="required"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="v-input v-text-field v-text-field--enclosed v-text-field--outline v-select theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Email Template
                  </label>
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-label="Email Template"
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      required="required"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                Template One
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                Template Two
                              </div>
                            </div>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="v-input v-text-field v-text-field--enclosed v-text-field--outline theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Email Subject
                  </label>
                  <input
                    aria-label="Email Subject"
                    required="required"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <!---->
           
          <div
            class="layout justify-end row"
          >
            <button
              class="v-btn v-btn--large v-btn--round theme--light"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
              Cancel
            
              </div>
            </button>
             
            <button
              class="v-btn v-btn--large v-btn--round theme--light"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
              Send
            
              </div>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
